solr/solrj/src/java/org/apache/solr/client/solrj/impl/InputStreamResponseParser.java

```diff
@@ -38,14 +43,35 @@ public InputStreamResponseParser(String writerType) {
     this.writerType = writerType;
   }
 
+  /**
+   * When using a {@link InputStreamResponseParser}, the raw output is available in the response
+   * under the key {@link #STREAM_KEY}.
+   */
+  public static String consumeResponseToString(NamedList<Object> response) throws IOException {
+    assert response != null;
+    String output;
+    // Would be nice to validate the STREAM_KEY value is present
+    try (InputStream responseStream = (InputStream) response.get(STREAM_KEY)) {
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      responseStream.transferTo(baos);
+      output = baos.toString(StandardCharsets.UTF_8);
+    }
+    return output;
+  }
+
   @Override
   public String getWriterType() {
     return writerType;
   }
 
   @Override
-  public NamedList<Object> processResponse(InputStream body, String encoding) {
-    throw new UnsupportedOperationException();
+  public NamedList<Object> processResponse(InputStream body, String encoding) throws IOException {
```

Comment:
I can't believe I let this slip through my code review; I need to review more carefully.  Clearly this change should not have been done.